src/main/java/org/elasticsearch/index/translog/fs/FsTranslog.java

```diff
@@ -162,20 +205,25 @@ public void close() throws IOException {
 
     @Override
     public long currentId() {
-        FsTranslogFile current1 = this.current;
-        if (current1 == null) {
-            return -1;
+        try (ReleasableLock _ = readLock.acquire()) {
```

Comment:
we acquire the read lock every indexing operation (also today). I don't think the performance implications are relevant. I chose to keep things as simple as possible. This is mostly relevant to the other places were it's harder (though not impossible) to by pass.